sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterMetricExtensions.cs

```diff
@@ -28,11 +28,26 @@ public static MeterProviderBuilder AddAzureMonitorMetricExporter(this MeterProvi
                 throw new ArgumentNullException(nameof(builder));
             }
 
-            var options = new AzureMonitorExporterOptions();
-            configure?.Invoke(options);
+            if (builder is IDeferredMeterProviderBuilder deferredMeterProviderBuilder)
```

Comment:
Can we follow up with @CodeBlanch about this?
I was under the impression that OTel was moving away from the IDeferredProviderBuilder.

See also: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.InMemory/InMemoryExporterMetricsExtensions.cs

Reply:
I see, will get this fixed. Will follow the recommendation from here - https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace/extending-the-sdk#tracerproviderbuilder-extension-methods

Reply:
So what is on this PR will work with the 1.3 API and continue to work with the 1.4 API but doesn't get you all the way there. It is also very ugly 🤣 The 1.4 API allows for a deeper more standard options integration without reflection in a little cleaner style.

Can merge this and update to the 1.4 API later or wait for it either works for me.